lib/BaseClient.ts

```diff
@@ -36,6 +39,26 @@ abstract class BaseClient extends EventEmitter {
     this.logger.info(`${this.constructor.name} status: ${ClientStatus[status]}`);
     this.status = status;
   }
+
+  /**
+   * Sends payment according to the terms of a swap deal.
+   * @returns the preimage for the swap
+   */
+  public abstract async sendPayment(deal: SwapDeal): Promise<string>;
+
+  /**
+   * Gets routes for the given currency, amount and peerPubKey.
+   * @param amount the capacity of the route
+   * @param destination target node for the route
+   * @returns routes
+   */
+  public abstract async getRoutes(amount: number, destination: string): Promise<Route[]>;
```

Comment:
In the case of raiden there's no `getRoutes` API, but if it were to be implemented it would also require either `contract_address` or `currency` in addition to `amount` and `destination`. For now it's fine, though.